app/models/rule.rb

```diff
@@ -53,8 +53,18 @@ class Rule < ActiveRecord::Base
   validates :name, presence: true
   validates :topic, presence: true, inclusion: TOPICS.keys
 
+  serialize :events, RuleEventRepositorySerializer
+
+  after_initialize :default_rule_event_repository
```

Comment:
Why is this required? Wouldn't empty events still return a repository given `      return RuleEventRepository.new if string.nil?`?

Reply:
You are right! I didn't know serializers were run on initialize